### PR TITLE
Add nonce / gas_price validity checking logic

### PR DESCRIPTION
**Pull request by platfowner**
Summary:
- Move default gas_price setting logic from client to node (createSingleTransaction())
- Add nonce / gas_price validity checking logic to isValidTxBody() and test cases
- Always use env variable instead of feature flags (remove feature flags) 
- Simplify gas price handling logic 

Related issue: https://github.com/ainblockchain/ain-blockchain/issues/316

### CODE REVIEW

File: tx-pool/transaction.js [hunk 1]
```diff
@@ -186,12 +187,31 @@ class Transaction {
...
 
   static hasRequiredFields(txBody) {
-    return txBody && txBody.timestamp !== undefined && txBody.nonce !== undefined &&
-        txBody.operation !== undefined;
+    return txBody && txBody.operation !== undefined && txBody.timestamp !== undefined &&
+        txBody.nonce !== undefined;
+  }
+
+  static isValidNonce(nonce) {
+    return ChainUtil.isInteger(nonce) && (nonce >= 0 || nonce === -1 || nonce === -2);
```
Comment (liayoo) [inline tx-pool/transaction.js:209]: question) Is this chosen over `ChainUtil.isInteger(nonce) && nonce >= -2` for readability?

**Review by liayoo — APPROVED** [review]
LGTM with a question. Thanks for the nice refactoring! 👍

File: tx-pool/transaction.js (hunk 1 above)
Comment (platfowner) [reply tx-pool/transaction.js]: That looks better. Just applied. Thanks.

**Comment by platfowner** [comment]
Thanks for the fast review!

**Pull request merged**